src/script_compiler.rs

```diff
@@ -29,15 +37,87 @@ extern "C" {
 #[derive(Debug)]
 pub struct Source([usize; 8]);
 
+/// Compilation data that the embedder can cache and pass back to speed up future
+/// compilations. The data is produced if the CompilerOptions passed to the compilation
+/// functions in ScriptCompiler contains produce_data_to_cache = true. The data to cache
+/// can then can be retrieved from UnboundScript.
+#[repr(C)]
+#[derive(Debug)]
+pub struct CachedData<'a> {
+  data: *const u8,
+  length: i32,
+  rejected: bool,
+  buffer_policy: BufferPolicy,
+  _phantom: PhantomData<&'a ()>,
```

Comment:
Can you add a static_assert to `src/binding.cc` that checks that the Rust and C++ definitions agree on the struct size?